database/gormdb_enrollment.go

```diff
@@ -88,6 +88,12 @@ func (db *GormDB) getEnrollments(model interface{}, statuses ...qf.Enrollment_Us
 		Find(&enrollments); err != nil {
 		return nil, err
 	}
+	for _, enrollment := range enrollments {
+		if enrollment.GetCourse() == nil {
```

Comment:
I don't think this check is necessary. If it is necessary, we should move the check inside the `SetSlipDays` method, and simply return if the provided course is `nil`.